### PR TITLE
Fix auto-fixable trunk errors

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

- Fixed auto fixable trunk errors (& a few more) to get the ball rolling
- Introduced git-blame-ignore-revs file (will add this merge commit hash to this file once this PR has been merged to prevent clogging up the blame)

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [x] Documentation update
- [x] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: tests/routes/test_document_families.py [hunk 1]
```diff
@@ -728,11 +727,9 @@ def test_update_document__logs_warning_on_four_letter_language(
     assert json_object["languages"] == []
 
     assert (
-        log_spy.call_args_list[0].args[0]
-        == "Retrieved no language from database for meta_data object "
-        "language"
+        "Retrieved no language from database for meta_data object language"
+        in caplog.text
     )
-    assert len(log_spy.call_args_list) == 1
```
Comment (THOR300) [inline tests/routes/test_document_families.py:735]: So this looks like actual code changes not just refactoring, was this introduced by trunq?

Comment (katybaulch) [reply tests/routes/test_document_families.py:735]: Indeed introduced by trunk - the error this fixed was specifically:
`715:22  high  Cannot access member "spy" for type "function" ↵   Member "spy" is unknown  pyright/reportFunctionMemberAccess`

File: app/core/download.py [hunk 2]
```diff
@@ -249,14 +250,17 @@ def get_whole_database_dump(ingest_cycle_start: str, db=Depends(get_db)):
 def replace_slug_with_qualified_url(
     df: pd.DataFrame,
     public_app_url: str,
-    url_cols: list[str] = ["Family Slug", "Document Slug"],
+    url_cols: Optional[list[str]] = None,
```
Comment (THOR300) [inline app/core/download.py:253]: Not a blocker just wondering why this is better!

File: docs/api/search.md [hunk 3]
```diff
@@ -93,22 +100,26 @@ This is an array containing exactly two values, which can be null or an integer
...
-## **Response Payload**
+The response returns a list of families and includes their associated documents
+along with their passage matches. The payload has the following scheme:
 
-The response returns a list of families and includes their associated documents along with their passage matches. The payload has the following scheme:
-
-```
+```json
 {
   "hits": 0,
```
Comment (THOR300) [inline docs/api/search.md:124]: This appears to show up in red as does the json below?: 

image

File: pyproject.toml [hunk 4]
```diff
@@ -3,6 +3,7 @@ name = "navigator_backend"
 version = "1.8.2"
 description = ""
 authors = ["CPR-dev-team <tech@climatepolicyradar.org>"]
+packages = [{ include = "app" }, { include = "tests" }]
```
Comment (THOR300) [inline pyproject.toml:6]: I don't this is technically necessary as both packages exist at the project root.

File: scripts/validate_cclw_sheet/validate-ingest.sh [hunk 5]
```diff
@@ -12,36 +12,36 @@ PASSWORD=${SUPERUSER_PASSWORD:-password}
...
+		"${TEST_HOST}"/api/v1/admin/bulk-ingest/validate/cclw
 }
 
 echo "Validating as ${USER}"
 wait_for_server
 
 echo
 echo "👉👉👉  Validate CSV"
-validate_csv > validation.json
+validate_csv >validation.json
```
Comment (THOR300) [inline scripts/validate_cclw_sheet/validate-ingest.sh:44]: Interesting!

**Review by THOR300 — APPROVED** [review]
LGTM - no blocking problems from me. This goes a long way to improving code quality!

File: app/core/download.py (hunk 2 above)
Comment (katybaulch) [reply app/core/download.py:253]: Because we shouldn't be using mutable data structures for argument defaults!

File: pyproject.toml (hunk 4 above)
Comment (katybaulch) [reply pyproject.toml:6]: It unfortunately is very necessary for local development, without this, you will get import errors using Trunk with a virtual environment (whether that's a venv or a pyenv)

File: scripts/validate_cclw_sheet/validate-ingest.sh
Comment (diversemix) [inline scripts/validate_cclw_sheet/validate-ingest.sh:1]: This file can get removed now

**Review by diversemix — APPROVED** [review]
lgtm

File: scripts/validate_cclw_sheet/validate-ingest.sh (hunk 5 above)
Comment (diversemix) [reply scripts/validate_cclw_sheet/validate-ingest.sh:44]: This removes the ambiguity of the space being a part of the filename

File: app/core/download.py (hunk 2 above)
Comment (THOR300) [reply app/core/download.py:253]: Nice!

File: scripts/validate_cclw_sheet/validate-ingest.sh
Comment (katybaulch) [reply scripts/validate_cclw_sheet/validate-ingest.sh:1]: I'll raise a separate ticket to sort out the scripts folder - I think they can probably go to navigator scripts

File: docs/api/search.md (hunk 3 above)
Comment (katybaulch) [reply docs/api/search.md:124]: Hmmm, I can't see that 🤔

Comment (katybaulch) [reply docs/api/search.md:124]: image

**Pull request merged**